msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java

```diff
@@ -3167,45 +3162,23 @@ public String processAddAttachmentRedirect()
   }
   
   //Process remove attachment 
-  public String processDeleteAttach()
-  {
+  public String processDeleteAttach() {
     log.debug("processDeleteAttach()");
-    
+
     ExternalContext context = FacesContext.getCurrentInstance().getExternalContext();
-    String attachId = null;
-    
+
     Map paramMap = context.getRequestParameterMap();
-    Iterator<Entry<Object, String>> itr = paramMap.entrySet().iterator();
-    while(itr.hasNext())
-    {
-      Entry<Object, String> entry = itr.next();
-      Object key = entry.getKey();
-      if( key instanceof String)
-      {
-        String name =  (String)key;
-        int pos = name.lastIndexOf("pvmsg_current_attach");
-        
-        if(pos>=0 && name.length()==pos+"pvmsg_current_attach".length())
-        {
-          attachId = entry.getValue();
-          break;
-        }
-      }
-    }
-    
+
+    final String attachId = (String) CollectionUtils.emptyIfNull(paramMap.entrySet()).stream().
```

Comment:
all this casting here why not just use generics?
Also the CollectionUtils.emptyIfNull is probably not needed if you use java lambda Collection

Reply:
You are right about the generics. I made a mistake with parenthesis in "CollectionUtils.emptyIfNull", I tried to avoid a NullPointerException obtaining the "entrySet" from the "paramMap".
Both solved!